File: atto-node/src/test/kotlin/org/atto/node/Waiter.java
package org.atto.node;

import java.util.concurrent.Callable;
import java.util.concurrent.TimeUnit;

import static org.awaitility.Awaitility.await;
import static org.hamcrest.Matchers.notNullValue;

public class Waiter {
    public static int timeoutInSeconds = 50000;

    public static <T> T waitUntilNonNull(Callable<T> callable) {
        return await().atMost(timeoutInSeconds, TimeUnit.SECONDS)
                .with()
                .pollDelay(50, TimeUnit.MILLISECONDS)
                .until(callable, notNullValue());
    }
}

File: atto-node/src/test/kotlin/org/atto/node/Waiter.kt
package org.atto.node

import org.awaitility.Awaitility
import org.hamcrest.Matchers
import java.util.concurrent.Callable
import java.util.concurrent.TimeUnit

object Waiter {
    var timeoutInSeconds = 60

    init {
        val isGradle = System.getenv("GRADLE")?.toBoolean() ?: false

        if (!isGradle) {
            timeoutInSeconds = 600
        }
    }

    fun <T> waitUntilNonNull(callable: Callable<T>?): T {
        return Awaitility.await().atMost(timeoutInSeconds.toLong(), TimeUnit.SECONDS)
            .with()
            .pollDelay(50, TimeUnit.MILLISECONDS)
            .until(callable, Matchers.notNullValue())
    }
}